src/contexts/states/logFileStore.ts

```diff
@@ -0,0 +1,150 @@
+import * as Comlink from "comlink";
+import {create} from "zustand";
+
+import {Nullable} from "../../typings/common";
+import {CONFIG_KEY} from "../../typings/config";
+import {LOG_LEVEL} from "../../typings/logs";
+import {
+    DO_NOT_TIMEOUT_VALUE,
+    LONG_AUTO_DISMISS_TIMEOUT_MILLIS,
+} from "../../typings/notifications";
+import {QueryResults} from "../../typings/query";
+import {UI_STATE} from "../../typings/states";
+import {SEARCH_PARAM_NAMES} from "../../typings/url";
+import {
+    CursorType,
+    FileSrcType,
+} from "../../typings/worker";
+import {getConfig} from "../../utils/config";
+import {updateWindowUrlSearchParams} from "../UrlContextProvider";
+import useContextStore from "./contextStore";
+import useLogExportStore, {LOG_EXPORT_STORE_DEFAULT} from "./logExportStore";
+import useLogFileManagerStore from "./LogFileManagerProxyStore";
+import useQueryStore from "./queryStore";
+import useUiStore from "./uiStore";
+import useViewStore, {VIEW_STORE_DEFAULT} from "./viewStore";
+
+
+interface LogFileValues {
+    fileName: string;
+    fileSrc: Nullable<FileSrcType>;
+    numEvents: number;
+    onDiskFileSizeInBytes: number;
+}
+
+interface LogFileActions {
+    setFileName: (newFileName: string) => void;
+    setNumEvents: (newNumEvents: number) => void;
+    setOnDiskFileSizeInBytes: (newOnDiskFileSizeInBytes: number) => void;
+
+    loadFile: (fileSrc: FileSrcType, cursor: CursorType) => void;
+}
+
+type LogFileState = LogFileValues & LogFileActions;
+
+const LOG_FILE_STORE_DEFAULT: LogFileValues = {
+    fileName: "",
+    fileSrc: null,
+    numEvents: 0,
+    onDiskFileSizeInBytes: 0,
+};
+
+/**
+ * Post a popup about the format string option in the settings.
+ */
+const postFormatPopup = () => {
+    useContextStore.getState().postPopUp({
+        level: LOG_LEVEL.INFO,
+        message: "Adding a format string can enhance the readability of your" +
+                    " structured logs by customizing how fields are displayed.",
+        timeoutMillis: LONG_AUTO_DISMISS_TIMEOUT_MILLIS,
+        title: "A format string has not been configured",
+    });
+};
+
+// eslint-disable-next-line max-lines-per-function
+const useLogFileStore = create<LogFileState>((set, get) => ({
+    ...LOG_FILE_STORE_DEFAULT,
+    loadFile: (fileSrc: FileSrcType, cursor: CursorType) => {
```

Comment:
Can we extract functions `onExportChunk` and `onQueryResults` from the body of `loadFile` into the global scope? Also, according to our handlers naming convention, they should be called `handleExportChunk` and `handleQueryResults`, right?